### PR TITLE
[master] Improve Homebrew's casks support

### LINKED ISSUE #66291: [BUG] MacOS: pkg.latest_version in mac_brew_pkg errors on cask lookup in official repo
**Description**
`latest_version` in `mac_brew_pkg.py` returns an error when trying to check version for casks in the official homebrew/cask repo. https://github.com/saltstack/salt/blob/v3007.0/salt/modules/mac_brew_pkg.py#L331C1

This does not occur with casks in other taps. https://github.com/cdalvaro/homebrew-tap/blob/main/Casks/salt.rb works fine.
 
**Setup**
v3007.0 on Apple Silicon using the package at https://repo.saltproject.io/salt/py3/macos/latest/salt-3007.0-py3-arm64.pkg
minion connected to remote master
`brew tap cdalvaro/tap`
`brew install --cask cdalvaro/tap/salt`

- [x] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
`salt-call pkg.latest_version tunnelblick` some cask from homebrew/cask
or
`salt-call state.apply test=True` with any cask from official repo in included states

```
% sudo salt-call pkg.latest_version homebrew/cask/tunnelblick
[ERROR   ] An un-handled exception was caught by Salt's global exception handler:
StopIteration: 
Traceback (most recent call last):
  File "/opt/salt/salt-call", line 23, in <module>
    sys.exit(salt_call())
  File "/opt/salt/lib/python3.10/site-packages/salt/scripts.py", line 443, in salt_call
    client.run()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/call.py", line 50, in run
    caller.run()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/caller.py", line 95, in run
    ret = self.call()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/caller.py", line 202, in call
    ret["return"] = self.minion.executors[fname](
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 160, in __call__
    ret = self.loader.run(run_func, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1233, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1248, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/executors/direct_call.py", line 10, in execute
    return func(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 160, in __call__
    ret = self.loader.run(run_func, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1233, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1248, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/modules/mac_brew_pkg.py", line 331, in latest_version
    return next(iter(versions_dict.values()))
StopIteration
Traceback (most recent call last):
  File "/opt/salt/salt-call", line 23, in <module>
    sys.exit(salt_call())
  File "/opt/salt/lib/python3.10/site-packages/salt/scripts.py", line 443, in salt_call
    client.run()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/call.py", line 50, in run
    caller.run()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/caller.py", line 95, in run
    ret = self.call()
  File "/opt/salt/lib/python3.10/site-packages/salt/cli/caller.py", line 202, in call
    ret["return"] = self.minion.executors[fname](
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 160, in __call__
    ret = self.loader.run(run_func, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1233, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1248, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/executors/direct_call.py", line 10, in execute
    return func(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 160, in __call__
    ret = self.loader.run(run_func, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1233, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/loader/lazy.py", line 1248, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/opt/salt/lib/python3.10/site-packages/salt/modules/mac_brew_pkg.py", line 331, in latest_version
    return next(iter(versions_dict.values()))
StopIteration
```
**Expected behavior**
The latest available version should be returned, as it is for casks from other taps.
```
% sudo salt-call pkg.latest_version cdalvaro/tap/salt
local:
    3007.0
```

**Screenshots**
N/A

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3007.0
 
Python Version:
        Python: 3.10.13 (main, Feb 19 2024, 03:29:13) [Clang 15.0.0 (clang-1500.0.40.1)]
 
Dependency Versions:
          cffi: 1.16.0
      cherrypy: 18.8.0
      dateutil: 2.8.2
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.3
       libgit2: Not Installed
  looseversion: 1.3.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.7
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 23.1
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.19.1
        pygit2: Not Installed
  python-gnupg: 0.5.2
        PyYAML: 6.0.1
         PyZMQ: 25.1.2
        relenv: 0.15.1
         smmap: Not Installed
       timelib: 0.3.0
       Tornado: 6.3.3
           ZMQ: 4.3.4
 
Salt Package Information:
  Package Type: onedir
 
System Versions:
          dist: darwin 22.6.0 
        locale: utf-8
       machine: arm64
       release: 22.6.0
        system: Darwin
       version: 13.6.3 arm64
 
```
</details>

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by cdalvaro**
### What does this PR do?

This PR allows to pass additional options to some methods in order to avoid ambiguities between formulae and casks.

Other improvements have been done:

- Fixes #66291
- Add support for aliases when getting info about packages

### Merge requirements satisfied?

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices, including the
[PR Guidelines](https://docs.saltproject.io/en/master/topics/development/pull_requests.html).

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cdalvaro** [comment]
ping @felippeb

File: salt/modules/mac_brew_pkg.py [hunk 1]
```diff
@@ -353,6 +370,10 @@ def remove(name=None, pkgs=None, **kwargs):
         A list of packages to delete. Must be passed as a python list. The
         ``name`` parameter will be ignored if this option is passed.
 
+    options
```
Comment (Akm0d) [inline salt/modules/mac_brew_pkg.py:373]: Please indicate that this new argument is expected to be a list of strings

**Review by Akm0d — DISMISSED** [review]

File: salt/modules/mac_brew_pkg.py (hunk 1 above)
Comment (cdalvaro) [reply salt/modules/mac_brew_pkg.py:375]: Done!

Comment (cdalvaro) [reply salt/modules/mac_brew_pkg.py:375]: Hi @Akm0d, I already did the requested changes.

Please, could you check changes are OK?

Thank you!

**Comment by cdalvaro** [comment]
ping @Akm0d @felippeb

**Review by twangboy — APPROVED** [review]

**Pull request merged**